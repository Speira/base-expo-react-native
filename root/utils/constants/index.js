// utils/constants
// description: We try to set all constants into a big one to reduce Javascript
//              constants performance issue.

export default {
  API: {
    FETCH_DATA_URL:
      'https://static.leboncoin.fr/img/shared/technical-test.json',
    PROXY: {
      NO_CORS_SERVER: 'https://cors-anywhere.herokuapp.com/',
    },
    TIMEOUT: 20000,
  },
  LABELS: {
    CLEAR_DATABASE: 'Clear the database',
    START_DATABASE: 'Create the database',
    IMVALID_DATA_FORMAT: ' Invalid data format',
    LIST: 'Item List',
    NO_CARD: 'No card to persist',
    NO_DATABASE: 'The database is not set',
    NO_RESPONSE_DATA: 'There is no data in the query response',
    REFETCH_DATA: 'Refetch Data',
    SEARCH_DATA_FAILED:
      'No data for SearchDataAPI: Please unlock the proxy below and then refetch data',
    WELCOME: 'Welcome to the ListItem app',
  },
  STATES: {
    ERROR: 'error',
    LOADING: 'loading',
  },
  STATUS: {
    DANGER: 'DANGER',
    INFO: 'INFO',
    SUCCESS: 'SUCCESS',
    WARNING: 'WARNING',
  },
  STORAGE_ITEMS: {
    DB: 'storage_list_item_app',
    CARD: 'card_items.db',
    THEME: 'app_theme_storage',
  },
  THEMES: {
    DEFAULT: {
      COLORS: {
        STATIC: {
          PRIMARY: '#EBD8C3',
          SECONDARY: '#F7E9D7',
          TERTIARY: '#FFF6EA',
          QUATERNARY: '#FFEEEE',
          LIGHT: '#f8f8fa',
          DARK: '#4d3e30',
        },
        DYNAMIC: {
          INFO: '#035397',
          SUCCESS: '#1C7947',
          WARNING: '#C36A2D',
          DANGER: '#A9333A',
        },
      },
    },
  },
}
