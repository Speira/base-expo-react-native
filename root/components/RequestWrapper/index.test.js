import React from 'react'
import { shallow } from 'enzyme'
import constants from '~utils/constants'
import RequesWrapper from './index'

import BaseRequestWrapper, {
  RequestError,
  RequestErrorWrapper,
  RequestLoader,
  RequestLoaderWrapper,
} from './style'

const { THEMES } = constants

describe('render RequesWrapper', () => {
  const wrapper = shallow(
    <RequesWrapper errors={[]} isLoading theme={THEMES.DEFAULT}>
      test
    </RequesWrapper>,
  )
  test('RequesWrapper should be defined', () => {
    expect(wrapper).toBeDefined()
  })
  test('RequesWrapper should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  test('RequesWrapper should have RequestLoader displayed ', () => {
    const loading = wrapper.find(RequestLoader)
    expect(loading).toHaveLength(1)
  })
  test('RequesWrapper can have RequestError displayed ', () => {
    let error = wrapper.find(RequestError)
    expect(error).toHaveLength(0)
    wrapper.setProps({ errors: [{ message: 'test' }], isLoading: false })
    error = wrapper.find(RequestError)
    expect(error).toHaveLength(1)
  })
  test('RequesWrapper can have the Request displayed ', () => {
    let request = wrapper.find(BaseRequestWrapper)
    expect(request).toHaveLength(0)
    wrapper.setProps({ errors: [] })
    request = wrapper.find(BaseRequestWrapper)
    expect(request).toHaveLength(1)
  })
})
