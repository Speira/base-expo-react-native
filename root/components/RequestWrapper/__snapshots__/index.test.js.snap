// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render RequesWrapper RequesWrapper should match snapshot 1`] = `
<style__RequestLoaderWrapper>
  <style__RequestLoader />
</style__RequestLoaderWrapper>
`;
